### DISCUSSION TITLE
Frequent need for --reauth

### DISCUSSION THREAD

**Discussion by frejohg** [abraunegg/onedrive#2410] [Q&A] [answered]
Hi

When attempting to sync, I often get the --reauth message, is this common?
This is a problem for me since I need a remote machine to run a bash script including a onedrive sync.

I wonder how it can be so? Could it be my university that has stricter policies?  In comparison dropbox never asks me to reauth.

**Comment by abraunegg** [comment, marked as answer]
@frejohg 
> When attempting to sync, I often get the --reauth message, is this common?

You generally need to perform a `--reauth` every time your account password changes.

> I wonder how it can be so? Could it be my university that has stricter policies? In comparison dropbox never asks me to reauth.

How often are you changing your University account password as this will be tied to your account that you are accessing OneDrive with.

> This is a problem for me since I need a remote machine to run a bash script including a onedrive sync.

So your actual problem could also be tied to this - as in, the application will be looking for the authentication token, and, it not present, or the client has issues reading it (thus cant use it) will generate an error similar to the following:
```
ERROR: Microsoft OneDrive API returned an error with the following message:
  Error Message:    HTTP request returned status code 400 (Bad Request)
  Error Reason:     AADSTS900144: The request body must contain the following parameter: 'refresh_token'.
Trace ID: 2644ba23-fedf-4e83-bd88-cd0f874c2700
Correlation ID: 41e63dea-e3b0-4760-a1b9-0fc30fb8ba1c
Timestamp: xxxxxxxxxxx
```
This sort of message can also generate the requirement to perform a `--reauth`.

My suggestion to you is to run the client in debug logging mode (https://github.com/abraunegg/onedrive/wiki/Generate-debug-log-for-support) so that you can actually understand the cause of your issue.
